pkg/operation/shoot/shoot.go

```diff
@@ -100,13 +100,29 @@ func (b *Builder) WithShootObjectFromLister(shootLister gardencorelisters.ShootL
 
 // WithCloudProfileObject sets the cloudProfileFunc attribute at the Builder.
 func (b *Builder) WithCloudProfileObject(cloudProfileObject *gardencorev1beta1.CloudProfile) *Builder {
-	b.cloudProfileFunc = func(string) (*gardencorev1beta1.CloudProfile, error) { return cloudProfileObject, nil }
+	b.cloudProfileFunc = func(context.Context, string) (*gardencorev1beta1.CloudProfile, error) { return cloudProfileObject, nil }
 	return b
 }
 
-// WithCloudProfileObjectFromLister sets the cloudProfileFunc attribute at the Builder after fetching it from the given lister.
-func (b *Builder) WithCloudProfileObjectFromLister(cloudProfileLister gardencorelisters.CloudProfileLister) *Builder {
-	b.cloudProfileFunc = func(name string) (*gardencorev1beta1.CloudProfile, error) { return cloudProfileLister.Get(name) }
+// WithCloudProfileObjectFromAPIReader sets the cloudProfileFunc attribute at the Builder after fetching it from the
+// given API reader.
+func (b *Builder) WithCloudProfileObjectFromAPIReader(reader client.Reader) *Builder {
```

Comment:
Doesn't have to be an **API**Reader, right? You can also pass a cache, right?
```suggestion
func (b *Builder) WithCloudProfileObjectFromReader(reader client.Reader) *Builder {
```

Reply:
Yep, done